.github/workflows/test-coverage.yml

```
@@ -0,0 +1,49 @@
+          restore-keys: ${{ runner.os }}-gradle-
+
+      # Build and run tests with Gradle
+      - name: Build and Test
+        run: ./gradlew build jacocoTestReport --info
+
+      # Run SonarQube analysis
+      - name: SonarQube Analysis
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew sonarqube --info
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)